util/query.py

```diff
@@ -327,7 +333,7 @@ def getNFTVolumes(
 def _filterOutPurgatory(nftvols: dict, chainID: int) -> dict:
```

Comment:
It appears that _filterOutPurgatory() was broken before your changes, since at each iteration of the loop it's updating the whole entry for basetoken, which could be a lot of NFTs. 

So this needs to get fixed, where it only updates the entry inside the basetoken dict. You can do that as part of this PR. I see that you're doing something like that in the new function _filterOutMarketAssets(). Great, use that pattern here.